src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const temporaryObj = { ...state };
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    if (type === 'addProperties') {
+      addProperties(extraData);
+    }
+
+    if (type === 'removeProperties') {
+      removeProperties(keysToRemove);
+    }
+
+    if (type === 'clear') {
+      clearProperties();
+    }
+
+    stateHistory.push({ ...temporaryObj });
```

Comment:
Pushing `{ ...temporaryObj }` to `stateHistory` will not work as intended because `temporaryObj` is not reset for each action. This will result in all entries in `stateHistory` being the same. Ensure that `temporaryObj` is a fresh clone of `state` at the start of each iteration.